src/services/announcement-service.ts

```diff
@@ -0,0 +1,206 @@
+    }
+
+    if (user.canvas.token === undefined) {
+      // There is no token
+      console.error('No token defined')
+      return undefined;
+    }
+    const canvas = (await db.collection(Collections.canvas).doc(canvasInstanceID).get()).data();
+    if (!canvas) {
+      return undefined;
+    }
+    /*
+      // # This call has a problem where it will only show the last 14 days or since a start date 
+      // # (which presumely) can't be before the course creation date.
+      // # Discussion topics never show announcements unless you for some reason ask to only show announcements.
+
+    return Axios.request<CanvasAnnouncement[]>({
+      headers: {
+        Authorization: `Bearer ${user.canvas.token}`
+      },
+      params: {
+        context_codes: ['course_' + req.params.courseID],
+        start_date: '2021-01-01'
+      },
+
+      method: 'GET',
+      baseURL: canvas.endpoint,
+      url: '/api/v1/announcements'
+    }).then((d) => res.send(d.data))
+      .catch((err) => res.sendStatus(err.status));
+    */
+    return Axios.request<CanvasAnnouncement[]>({
+      headers: {
+        Authorization: `Bearer ${user.canvas.token}`
+      },
+      params: {
+        only_announcements: 'true'
+      },
+
+      method: 'GET',
+      baseURL: canvas.endpoint,
+      url: `/api/v1/courses/${courseID}/discussion_topics`
+    }).then((d) => d.data)
+      .catch(() => undefined);
+    //TODO: handle refresh/ 401/403
+  }
+
+  static async buildAnnouncementEmbed(announcement: CanvasAnnouncement, courseID: string, canvasInstanceID: string, discordUserID: string): Promise<MessageEmbed> {
+    const ts = new TurndownService();
+
+    const courses = await CanvasController.getCourses(discordUserID, canvasInstanceID);
+    if (courses === undefined) {
+      throw new Error('Courses not defined. Likely invalid or undefined token from users.');
+    }
+    const course = courses.find(c => c.id === parseInt(courseID));
+
+    const postedTime = new Date(announcement.posted_at);
+    const postTimeString = postedTime.getHours() + ':' + (postedTime.getMinutes() < 10 ? '0' + postedTime.getMinutes() : postedTime.getMinutes()) + ' • ' + postedTime.getDate() + '/' + (postedTime.getMonth() + 1) + '/' + postedTime.getFullYear();
+
+    const embed = new MessageEmbed({
+      color: '#E63F30',
+      title: announcement.title,
+      url: announcement.html_url,
+      author: {
+        name: course?.course_code + ' - ' + announcement.author.display_name
+      },
+      description: ts.turndown(announcement.message),
+      footer: {
+        text: postTimeString
+      }
+    });
+    return embed;
+  }
+
+  // ## Checking announcements for changes
+  // FIX: API checks before bot can receive events -> lastAnnounce is set to something that isn't posted
+  // TODO: check rate limits. Currently 1 minute interval. We want this as low as is allowed.
+  // guildID: string, CanvasInstanceID: string
+  static initAnnouncementJob(): NodeJS.Timeout {
```

Comment:
cleanup console.log()'s
refactor to be more readable

Reply:
Resolved in #21